File: src/main/java/com/store/service/OrderService.java
package com.store.service;

import com.store.dto.OrderDto;
import com.store.entity.User;

import java.util.List;
import java.util.Optional;

public interface OrderService {

    OrderDto createOrder(OrderDto orderDto, User user);

    List<OrderDto> getOrdersByUser(User user);

    Optional<OrderDto> getOrderById(Long id);

    List<OrderDto> getAllOrders();

    OrderDto updateOrderStatus(Long id, String status);

    // ✅ Добавлен метод, которого не хватало:
    void deleteOrder(Long id);
}

File: frontend/src/api/orderService.js
import api from './api';

export const orderService = {
  // Create a new order
  createOrder: (orderData) => {
    return api.post('/orders', orderData);
  },

  // Get user's orders
  getUserOrders: () => {
    return api.get('/orders/user');
  },

  // Get order by ID
  getOrderById: (id) => {
    return api.get(`/orders/${id}`);
  },

  // Get all orders (admin only)
  getAllOrders: () => {
    return api.get('/orders');
  },

  // Update order status (admin only)
  updateOrderStatus: (orderId, status) => {
    return api.put(`/orders/${orderId}/status`, { status });
  }
};
